django_prometheus/conf/__init__.py

```diff
@@ -0,0 +1,6 @@
+from django.conf import settings
+
+if not settings.configured:
+    settings.configure()
```

Comment:
This feels dangerous... we should not be calling this Django method from a library.
This can break applications import this file.

Reply:
so we can just put a default value in NAMESPACE if settings is not configured.

Reply:
@asherf, please let me set namespace from the environment, it's very useful

Reply:
by default, it will try to get it from settings and if it's not set - then try to get it from env, and only after that will set default

Reply:
namespace from the environment will override namespace in settings

Reply:
why do we need to be able to override with an environment variable at the library level ?
This is something that should be typically left to the app, it can decide where it wants to load settings from.

Reply:
I agree with @asherf on this.
The responsibility for this should be on the app and no in the library as then it would be difficult to change the behavior.
For the same reason that I need to be able to use the namespace, I might need the variable to be named differently (have a prefix for instance). It gets worse if I'm running several applications on the same environment and want to have a different namespace for each.

Reply:
I already changed that part

Reply:
> why do we need to be able to override with an environment variable at the library level ?
> This is something that should be typically left to the app, it can decide where it wants to load settings from.

yes, and you can decide
don't add this variable to the environment and it will be got from settings or set to default

Reply:
for me, it's easier to manage environment variables rather than settings